api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -126,4 +126,13 @@
    * @return the location of the manifest list for this Snapshot
    */
   String manifestListLocation();
+
+  /**
+   * Return the id of the schema used when this snapshot was created, or null if this information is not available.
+   *
+   * @return schema id associated with this snapshot
+   */
+  default Integer schemaId() {
+    return null;
```

Comment:
Yes, schemaId() returns null in the case where the snapshot was written before this change. Note though, that even after this change, new metadata can have snapshots without schema id (so schemaId() for those snapshots will return null), if it is metadata for a table existing before this change.

Reply:
I have a PR (https://github.com/apache/iceberg/pull/1508) that reads previous metadata to get the schema for the snapshot in case `Snapshot#schemaId()` returns null.

Reply:
Sorry for the delay in response, and thank you @openinx for the review! And thank you @wypoon for responding!